source/_posts/2015-10-09-the-slack-users-code.markdown

```diff
@@ -0,0 +1,258 @@
+channels are everything else.
+
+
+#### Minimize Use of Bots & Other Toys
+
+Programmers, especially, can get enamoured of chat bots and chat integrations.
+As a programmer, I feel this, too. However, you have to carefully balance this
+with the cost it can have on real human communication. Slack is a tool primarily
+for humans to communicate with other humans. If computers are being too chatty
+in a room, the humans will have a hard time getting a word in edge-wise. And a
+chat history tends to be a pretty poor radiator of information. Rather than
+having engineers initiate and control a deploy from the chat room, consider
+having a different tool that does all that, and have only sparse or summary
+updates go to the team room.
+
+As for other toys, let us consider the cautionary tale of `/giphy`. The images
+that [Giphy](http://giphy.com/) serves up in response to any given query are
+actually relevant to the query so infrequently that it's a remarkable event when
+it happens. I mean that literally: I will type into Slack "Holy shit. Giphy
+didn't fail miserably!" on the rare occasion.
+
+Some people feel that a single gif has a damaging effect on conversation. I
+_like_ a good animated gif, but people repeatedly typing `/giphy` over and over
+at each other hoping for a relevant animation does not constitute human
+interaction. This same principle applies to almost any fun integration or
+@slackbot auto-response. They're very easy to abuse and easy to do so in a way
+that you don't realize you're abusing them. No one should have to ask their
+co-workers to stop playing a loud game of tag in the hallway next to the
+bullpen.
+
+
+#### Archive Unused Channels
+
+If a channel is no longer needed, a Slack administrator (at any time) or the channel owner
+(if the channel is empty) should be able to archive it. Get rid of that dead
+weight.
+
+
+### Work-Related Guidelines
+
+#### Name it After the Hosting Team
+
+The channel structure should loosely mirror your company's org chart. Each
+channel should be hosted by a manager and their team. These channels should be
+named after the team. The Support team's channel should be #support and the
+API sub-team within should be #support-api (thus ordered so they sort together
+in the channel list).
+
+These channels are _hosted_ by a team and belong to that team, but they should
+still be public. If someone has a question about something owned by a team or
+for a member of that team, this channel is the first place they should go.
+
+Anyone is allowed to be in any channel, but members of a team are expected to
+be actively in their team's channel most of the time. In fact, if each manager
+hosts a Slack channel, everyone who reports up through them, even indirectly,
+should probably at least passively monitor every channel up the chain of
+command from them (imagine, e.g. a place where a VP level person can do
+"@channel: The code for the all-hands dial-in is 9466.").
```

Comment:
Might be good to include a hedge or caveat on “Anyone is allowed to be in any channel”, like “…but be aware of where your input is actively welcome vs. potentially counter-productive, e.g. in Holman’s discussion of HR policies.”

Something about being able to walk into any team’s physical room and observe their discussion, but having the social tact and self-awareness not to walk in and start telling them how to do their jobs.